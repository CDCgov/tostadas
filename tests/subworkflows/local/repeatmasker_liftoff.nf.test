nextflow_workflow {

    name "Test Workflow REPEATMASKER_LIFTOFF"
    script "subworkflows/local/repeatmasker_liftoff.nf"
    workflow "REPEATMASKER_LIFTOFF"

    test("Should run without failures") {

        when {
            params {
                outdir           = "${outputDir}"
                species          = "mpxv"
                ref_fasta_path   = "${projectDir}/assets/ref/ref.MPXV.NC063383.v7.fasta"
                ref_gff_path     = "${projectDir}/assets/ref/ref.MPXV.NC063383.v7.gff"
                repeat_library   = "${projectDir}/assets/lib/mpxv_repeats_lib.fasta"

            }
            workflow {
                """
                input[0] = Channel.of([
                    [ batch_id: "batch_1", batch_tsv: file("${projectDir}/tests/modules/local/prep_submission/batch_1.tsv"), sample_id: "NY0006" ],
                    file("${projectDir}/assets/sample_fastas/mpox/NY0006.fasta")
                ])
                """
            }
        }

        then {
            assert workflow.success
        }

    }

}
