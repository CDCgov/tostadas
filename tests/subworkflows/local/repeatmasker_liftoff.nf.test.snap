{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "batch_id": "batch_1",
                            "batch_tsv": "batch_1.tsv:md5,4ab00a8f04e89e82dd5504f5ca0c0a6c",
                            "sample_id": "NY0006"
                        },
                        "NY0006.fasta:md5,4a11c4fe6d828bf5e411c21d10075b7c"
                    ]
                ],
                "1": [
                    [
                        {
                            "batch_id": "batch_1",
                            "batch_tsv": "batch_1.tsv:md5,4ab00a8f04e89e82dd5504f5ca0c0a6c",
                            "sample_id": "NY0006"
                        },
                        "NY0006_reformatted.gff:md5,19697038c49dd5b9324bd8465bd91a2c"
                    ]
                ],
                "fasta": [
                    [
                        {
                            "batch_id": "batch_1",
                            "batch_tsv": "batch_1.tsv:md5,4ab00a8f04e89e82dd5504f5ca0c0a6c",
                            "sample_id": "NY0006"
                        },
                        "NY0006.fasta:md5,4a11c4fe6d828bf5e411c21d10075b7c"
                    ]
                ],
                "gff": [
                    [
                        {
                            "batch_id": "batch_1",
                            "batch_tsv": "batch_1.tsv:md5,4ab00a8f04e89e82dd5504f5ca0c0a6c",
                            "sample_id": "NY0006"
                        },
                        "NY0006_reformatted.gff:md5,19697038c49dd5b9324bd8465bd91a2c"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-28T16:33:00.022359022"
    }
}